No anonymous functions for scheduleOnce or once

eddie-ruva opened the pull request:
In this PR I'm adding a rule to tackle the issue brought up in #57. In this case I'm checking for function uniqueness by looking into the name property of the function and seeing if it is defined. 

This rule will only be checked when you call `Ember.run.scheduleOnce`, `run.scheduleOnce`, `Ember.run.once` and `run.once`. We could potentially also check for just calling `scheduleOnce` or `once` but wasn't to sure about which route we wanted to take this.

trentmwillis commented on lib/rules/no-anonymous-once.js:
```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { getCaller } = require('../utils/caller');
+const MESSAGE = 'The uniqueness once offers is based on function uniqueness, each invocation of this line will always create a new function instance, resulting in uniqueness checks, that will never be hit. Please replace with a named function.';
```
Can we add a link to the guide/docs for this rule as well?

eddie-ruva replied:
I was not sure if you meant the ember docs or the guides that exist inside this repo. Since the ember docs mention this, I added a the link to the `scheduleOnce` docs.

eddie-ruva commented:
@trentmwillis Thanks for taking a look and sorry for abandoning this PR for some time after you left the review. I've addressed your comment and also added code to cover for the case sin which  `scheduleOnce` and `once` are invoked with `call` and `apply`. Let me know what you think whenever you have some time. Thanks!
